Update Coverage.py

MaxBo opened the pull request:
this commit fixes #1898
use filepath with the correct upper and lowercases

scoder commented on Cython/Coverage.py:
```diff
@@ -45,6 +46,35 @@ def _find_dep_file_path(main_file, file_path):
+    # disk letter
+    test_name = [dirs[0].upper()]
+    for d in dirs[1:]:
+        test_name += ["%s[%s]" % (d[:-1], d[-1])]
```
How to append to a list? Easy! Create a new one and concatenate it to the existing one! :D
(I know, it's now your code...)

MaxBo replied:
Thats whats happes if you copy&paste random code ;-)
I fixed this

scoder commented on Cython/Coverage.py:
```diff
@@ -45,6 +46,35 @@ def _find_dep_file_path(main_file, file_path):
     return abs_path
 
 
+def _get_actual_filename(filepath):
```
I think this should just use `os.path.normcase()`.

MaxBo commented:
Fixed coverage tests in windows by renaming the generated pyd-files from `pkg\\coverage_test_py.cp36-win_amd64.pyd` to `pkg\\coverage_test_py.pyd`

Added MixedCase file- and packagenames to the coverage_api tests

scoder commented on Cython/Coverage.py:
```diff
@@ -9,9 +9,11 @@
 import re
 import os.path
 import sys
+import glob
```
Left-over import.

MaxBo replied:
deleted
